src/Switcher.java

```diff
@@ -0,0 +1,15 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class Switcher {
+
+    private Map<GameId, GameStrategy> gameList = new HashMap<>();
+
+    public void enterId(GameId id, GameStrategy strategy){
```

Comment:
you can use Integer instead of enum as identification for your game, another way is to use enums value instead of enum it self. 
Like: `GameId id.getId()`
by the way you can do all this in empty constructor...